monitoring/opentelemetry-reactive/src/test/java/io/quarkus/ts/opentelemetry/reactive/OpenTelemetryGrpcIT.java

```diff
@@ -26,7 +26,9 @@ public class OpenTelemetryGrpcIT {
     @QuarkusApplication()
     static RestService app = new RestService()
             .withProperty("quarkus.application.name", "pingpong")
-            .withProperty("quarkus.opentelemetry.tracer.exporter.otlp.endpoint", jaeger::getCollectorUrl);
+            .withProperty("quarkus.opentelemetry.tracer.exporter.otlp.endpoint", jaeger::getCollectorUrl)
+            .withProperty("quarkus.openshift.readiness-probe.grpc-action", "9000:io.quarkus.example.PongService")
+            .withProperty("quarkus.openshift.liveness-probe.grpc-action", "9000:io.quarkus.example.PongService");
```

Comment:
is it possible to specify also method? e.g. does this call `sayPong` or `returnLastTraceId`? and can we set different method for liveness and readiness probe so that we have verified it works when liveness != readiness?

Reply:
That is a good question.
According to the specification[1], a cluster should use Check method of the service automatically [2] and "unimplemented health checking protocol <...> considered a probe failure". I would say our cluster doesn't use grpc probes, but they are enabled by default since 1.24[3] and we use 1.25 and there were no complaints during deployment.

[1] https://github.com/grpc/grpc/blob/master/doc/health-checking.md
[2] https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe
[2] https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-graduated-or-deprecated-features

Reply:
So if you added SmallRye Health this gRPC Health service would be generated automatically for you right? Do we have a way to verify that gRPC probes were actually used? e.g. right now we check that template is there and we assume that pods are ready because they were not killed and we are running tests, but we don't check that template doesn't contain also HTTP probes that could be used?

Unlike you, I didn't test it so you can lead me, but I'd really like to:

- check that it fails when such Health service is not available
- check it doesn't fail when it is available and pod is ready and there are no other probes that could be used

Reply:
I verified, that the deployment fails, if the service is misspelled (eg `HelathService`). At the same time, I was not able to make deployment fail in because of error in liveness probe (see the code for an example). I will create a stand-alone reproducer and an issue on Tuesday

Reply:
Sounds good.

Reply:
The issue: https://github.com/quarkusio/quarkus/issues/33219